src/polly/src/event_manager.rs

```diff
@@ -52,7 +52,9 @@ pub trait Subscriber: Send {
     ///
     /// # Arguments
     /// * event - the available `EpollEvent` ready for processing
-    fn process(&mut self, event: EpollEvent);
+    /// * event_manager - Reference to the `EventManager` that gives the implementor
```

Comment:
Explicitly call out that one should avoid calling `.run()` or `.run_with_timeout()` on this reference as that can lead to deadlocks.
Obviously users also shouldn't do `.add_subscriber(self)` as that is a guaranteed deadlock.

As future work, it might be a fun exercise to have two layers to the `EventManager` struct and have only the top layer to implement `run()` while the second can implement the add/modify/unregister functions which are safe to call from `process()`. `process()` could then provide a `&mut SecondLayer`.

I think the documentation updates/warnings are enough for now.